tests/Aspire.Hosting.Tests/Nats/AddNatsTests.cs

```diff
@@ -0,0 +1,107 @@
+        Assert.Equal("tcp", endpoint.UriScheme);
+
+        var containerAnnotation = Assert.Single(containerResource.Annotations.OfType<ContainerImageAnnotation>());
+        Assert.Equal("2", containerAnnotation.Tag);
+        Assert.Equal("nats", containerAnnotation.Image);
+        Assert.Null(containerAnnotation.Registry);
+    }
+
+    [Fact]
+    public void AddNatsContainerAddsAnnotationMetadata()
+    {
+        var appBuilder = DistributedApplication.CreateBuilder();
+        appBuilder.AddNats("nats", 1234).WithJetStream(srcMountPath: "/tmp/dev-data");
+
+        var app = appBuilder.Build();
+
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+
+        var containerResource = Assert.Single(appModel.Resources.OfType<NatsServerResource>());
+        Assert.Equal("nats", containerResource.Name);
+
+        var manifestAnnotation = Assert.Single(containerResource.Annotations.OfType<ManifestPublishingCallbackAnnotation>());
+        Assert.NotNull(manifestAnnotation.Callback);
+
+        var mountAnnotation = Assert.Single(containerResource.Annotations.OfType<ContainerMountAnnotation>());
+        Assert.Equal("/tmp/dev-data", mountAnnotation.Source);
+        Assert.Equal("/data", mountAnnotation.Target);
+
+        var argsAnnotation = Assert.Single(containerResource.Annotations.OfType<CommandLineArgsCallbackAnnotation>());
+        Assert.NotNull(argsAnnotation.Callback);
+        var args = new List<string>();
+        argsAnnotation.Callback(new CommandLineArgsCallbackContext(args));
+        Assert.Equal("-js -sd /data".Split(' '), args);
+
+        var endpoint = Assert.Single(containerResource.Annotations.OfType<EndpointAnnotation>());
+        Assert.Equal(4222, endpoint.ContainerPort);
+        Assert.False(endpoint.IsExternal);
+        Assert.Equal("tcp", endpoint.Name);
+        Assert.Equal(1234, endpoint.Port);
+        Assert.Equal(ProtocolType.Tcp, endpoint.Protocol);
+        Assert.Equal("tcp", endpoint.Transport);
+        Assert.Equal("tcp", endpoint.UriScheme);
+
+        var containerAnnotation = Assert.Single(containerResource.Annotations.OfType<ContainerImageAnnotation>());
+        Assert.Equal("2", containerAnnotation.Tag);
+        Assert.Equal("nats", containerAnnotation.Image);
+        Assert.Null(containerAnnotation.Registry);
+    }
+
+    [Fact]
+    public void WithNatsContainerOnMultipleResources()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+        builder.AddNats("nats1");
+        builder.AddNats("nats2");
+
+        Assert.Equal(2, builder.Resources.OfType<NatsServerResource>().Count());
+    }
+
+    [Fact]
+    public async Task VerifyManifest()
+    {
+        var appBuilder = DistributedApplication.CreateBuilder();
+        var nats = appBuilder.AddNats("nats");
+
+        var manifest = await ManifestUtils.GetManifest(nats.Resource);
+
+        Assert.Equal("container.v0", manifest["type"]?.ToString());
```

Comment:
@mtmk we made some changes here https://github.com/dotnet/aspire/blob/f6a2f9f0b5aca2553aeab6d09ec657cffc30d9fd/tests/Aspire.Hosting.Tests/Kafka/AddKafkaTests.cs#L75-L95. 

BTW I'm sorry that these changes must feel like a roller coaster 😄. FWIW, it's helping us land on solid patterns that seem to scale well 🚀

Reply:
no worries, my fault really having the PR open so long. thanks for bearing with me 🙏 also loving to witness the evolution 😍